### PR TITLE
dev-libs/optix: Add optix v7.x

### PR DESCRIPTION
**Pull request by kshots**
Blender utilizes optix as an alternative to CUDA to utilize the
raytracing cores in the RTX series of video cards. version 7.0 has been
tested with Blender 2.8.4.

Package-Manager: Portage-3.0.4, Repoman-3.0.1

### CODE REVIEW

File: dev-libs/optix/optix-7.0.0.ebuild [hunk 1]
```diff
@@ -0,0 +1,37 @@
+# Copyright 2020 Gentoo Authors
```
Comment (epsilon-0) [inline dev-libs/optix/optix-7.0.0.ebuild:1]: Do we really need a `7.0.0` installation?
I would image a minor version bump should be backwards compatible, but then this *is* nvidia.

File: dev-libs/optix/optix-7.1.0.ebuild [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+SLOT="0/7"
+KEYWORDS="~amd64"
+RESTRICT="fetch"
+LICENSE="NVIDIA-r2"
+
+RDEPEND="dev-util/nvidia-cuda-toolkit
+	virtual/opengl
+	media-libs/freeglut"
+DEPEND="${RDEPEND}"
```
Comment (epsilon-0) [inline dev-libs/optix/optix-7.1.0.ebuild:18]: Remove `DEPEND`, installation is just an extraction, no need for any libraries.

File: dev-libs/optix/optix-7.1.0.ebuild [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+HOMEPAGE="https://developer.nvidia.com/optix"
+SRC_URI="NVIDIA-OptiX-SDK-${PV}-linux64-x86_64.sh"
+
+SLOT="0/7"
+KEYWORDS="~amd64"
+RESTRICT="fetch"
+LICENSE="NVIDIA-r2"
+
+RDEPEND="dev-util/nvidia-cuda-toolkit
+	virtual/opengl
```
Comment (epsilon-0) [inline dev-libs/optix/optix-7.1.0.ebuild:16]: Please order them alphabetically.

File: dev-libs/optix/optix-7.1.0.ebuild [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+src_unpack() {
+	tail -n +223 "${DISTDIR}"/${A} | tar -zx || die
+}
+
+src_install() {
+	insinto /opt/${PN}
+	dodoc -r doc
+	doins -r include SDK
+}
```
Comment (epsilon-0) [inline dev-libs/optix/optix-7.1.0.ebuild:37]: Otherwise look OK.

**Review by epsilon-0 — CHANGES_REQUESTED** [review]
Thanks for the update.

File: dev-libs/optix/optix-7.0.0.ebuild (hunk 1 above)
Comment (kshots) [reply dev-libs/optix/optix-7.0.0.ebuild:1]: media-gfx/blender-2.8.4 does not compile with dev-libs/optix-7.1.0, but does compile with dev-libs/optix-7.0.0. However, media-gfx/blender-2.9.0 compiles with dev-libs/optix-7.1.0. So, depending on which version of blender you want, you may need to stay < 7.1.0

File: dev-libs/optix/optix-7.1.0.ebuild (hunk 2 above)
Comment (kshots) [reply dev-libs/optix/optix-7.1.0.ebuild:18]: Agreed - when I first poked at this, I thought the SDK needed to be compiled. Then I discovered the installer doesn't compile it itself, so why should we?

File: dev-libs/optix/optix-7.1.0.ebuild (hunk 3 above)
Comment (kshots) [reply dev-libs/optix/optix-7.1.0.ebuild:16]: I'm a bit mystified here... order what alphabetically?

File: dev-libs/optix/optix-7.0.0.ebuild (hunk 1 above)
Comment (epsilon-0) [reply dev-libs/optix/optix-7.0.0.ebuild:1]: thanks nvidia...

File: dev-libs/optix/optix-7.1.0.ebuild (hunk 2 above)
Comment (epsilon-0) [reply dev-libs/optix/optix-7.1.0.ebuild]: Sorry, maybe I was unclear.
Keep the `RDEPENDS`, only remove the `DEPENDS`.
This is important for cross compilation.

Comment (kshots) [reply dev-libs/optix/optix-7.1.0.ebuild]: Oh! Putting them back now...

Comment (epsilon-0) [reply dev-libs/optix/optix-7.1.0.ebuild]: RDEPENDS = packages needed during running time
DEPEND = packages needed during linking time
BDEPEND = packages needed to be **run** to compile.

**Comment by epsilon-0** [comment]
You need to add a Sign-Off to the commit - https://wiki.gentoo.org/wiki/Talk:Gentoo_git_workflow#Sign-Off

Also squash the commits into one instead of so many.

**Comment by kshots** [comment]
Thanks for the link, rebased with --signoff enabled to combine all three commits into one.

**Comment by epsilon-0** [comment]
Thanks, committed.

**Pull request closed without merging**